src/commands/EmbedCommand.ts

```diff
@@ -0,0 +1,87 @@
+import { Command } from "../interactions/interactionClasses";
+import {
+	ActionRowBuilder,
+	ChatInputCommandInteraction,
+	ModalBuilder,
+	SlashCommandBuilder,
+	TextInputBuilder,
+	TextInputStyle,
+} from "discord.js";
+
+class EmbedCommand extends Command {
+	constructor() {
+		super("embed");
+	}
+
+	async execute(interaction: ChatInputCommandInteraction): Promise<void> {
+		const fields =
+			interaction.options
+				.getString("fields")
+				?.split(",")
+				.map((s) => s.trim())
+				.filter((s) => s)
+				.slice(0, 3) ?? [];
+		const showAuthor = interaction.options.getBoolean("show-author") ?? false;
+
+		const modal = new ModalBuilder()
+			.setCustomId(`embed.${showAuthor}${fields.length ? "." + fields.join(".") : ""}`)
```

Comment:
Would break if a field name has a dot in them, but do we care? A solution would be some encode (uri / base64) or the in-memory map.